batch/batch/aioclient.py

```diff
@@ -12,6 +12,15 @@
 
 
 class Job:
+    @staticmethod
+    def exit_code(job_status):
+        if 'exit_code' in job_status and job_status['exit_code'] is not None:
+            exit_codes = job_status['exit_code']
+            exit_codes = [exit_codes.get(task) for task in ['input', 'main', 'output']]
+            exit_codes = [ec for ec in exit_codes if ec is not None]
+            return exit_codes[-1] if len(exit_codes) != 0 else None
```

Comment:
Sorry, this should be the first non-zero value (if there is one), None (if there is one), or finally 0 if they are all zero.  Basically the exit code should be the first failure, e.g. we want the main exit code if main failed, even if the output step also failed.  (I know we don't run output unconditionally yet, but we want to.)

Reply:
Good suggestion. Should be fixed now.